### PR TITLE
Add support for stack

### PR DESCRIPTION
**Pull request by sigrlami**
- add support for stack
- bump version small number
- bump deps

@RyanGlScott I can support build with stack, just ping me after updates

### CODE REVIEW

File: keycode.cabal [hunk 1]
```diff
@@ -37,7 +37,7 @@ library
                      , containers
                      , ghc-prim
   if impl(ghc >= 8.0)
-    build-depends:     template-haskell >= 2.11 && < 2.13
+    build-depends:     template-haskell >= 2.11.1.0 && < 2.13
```
Comment (RyanGlScott) [inline keycode.cabal]: What is the purpose of this change?

**Review by RyanGlScott — COMMENTED** [review]
Hi @sigrlami.

I don't understand the point of this change. `keycode` should already work with `stack` (and indeed, it's currently in Stackage). Is there a reason you need this particular change?

File: keycode.cabal (hunk 1 above)
Comment (sigrlami) [reply keycode.cabal]: @RyanGlScott building with reflex-dom and reflex suite fails due shadow deps

Comment (sigrlami) [reply keycode.cabal]: I can keep it in my fork, just thought might be useful

Comment (sigrlami) [reply keycode.cabal]: also, there is a strange error in Firefox, but I didn't get to the bottom of it yet

Comment (RyanGlScott) [reply keycode.cabal]: OK... I don't have access to GHCJS, so I'm unable to reproduce this error you're claiming to experience. But I'm certainly not going to accept any change which excludes `template-haskell-2.11.0.0` from install plans, since that version of `template-haskell` is bundled with GHC 8.0.1.

Comment (sigrlami) [reply keycode.cabal]: @RyanGlScott no problem, thanks. Decline PR then, I'll keep it in my fork until I find a solution to keep 2.11.0.0

**Comment by RyanGlScott** [comment]
The one change from this PR I'm comfortable bringing in is adding `.stack-work` to `.gitignore`, which I've done in c4e815cf3f1257b2cbdc8471da5c80801fbd20ea. I don't ever check in `stack.yaml` files to my repos, since my workflow uses `cabal-install` rather than `stack.yaml`, so I'll drop that change.
